client/pages/StatUnits/View/StatUnitViewPage.jsx

```diff
@@ -89,6 +89,29 @@ class StatUnitViewPage extends React.Component {
     />
   )
 
+  handleCancel = () => {
+    this.setState({ showConfirm: false })
+  }
+
+  handleConfirm = () => {    
```

Comment:
Is it possible to have clearer names?
`handleConfirm` -> `confirmDelete`
`onConfirm` -> `performDelete`
`handleCancel` -> `cancelDelete`
`showConfirm` -> `showConfirmDelete`
`deleteFailed` -> Make it a string with the actual error message from the backend.